src/components/virtualRepeat/demoDeferredLoading/index.html

```diff
@@ -0,0 +1,22 @@
+<div ng-controller="AppCtrl as ctrl">
+  <md-content layout="column">
+    <p>
+       Display a list of 50,000 items that load on demand in a viewport of only 7 rows (height=40px).
+       <br/><br/>
+       This demo shows scroll and rendering performance gains when using <code>md-virtual-repeat</code>;
+       achieved with the dynamic reuse of rows visible in the viewport area. Developers are required to
+       explicitly use <code>md-virtual-repeat-container</code> as a wrapping parent container.
+       <br/><br/>
+       To enable load-on-demand behavior, developers must pass in a custom instance of
+       mdVirtualRepeatModel (see the example's source for more info).
```

Comment:
@jelbourn, @kseamon - I like the `md-on-demand` attribute and the use of the attribute instead of the model subclassing. Currently issues:

*  I do not think we are clearly articulating the [expected model interface methods](https://github.com/angular/material/pull/4002/files#diff-30883b316af6f89bca301e70a4243fbfR10) when `onDemand` is enabled; the comment is very easy to miss.
  * 
  *  Should this information also be in the source docs?
*  If the interface is not correct, I think we should throw an Error **or** at least warn the developer in the console.